src/Orchard.Web/Modules/Orchard.Users/Services/IUserManagementActionsProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Mvc;
+using Orchard.Security;
+
+namespace Orchard.Users.Services {
+    public interface IUserManagementActionsProvider : IDependency {
+        // Using a delegate so implementations don't have to build/figure out
+        // their own HtmlHelper
+        IEnumerable<Func<HtmlHelper, MvcHtmlString>> UserActionLinks(IUser user);
```

Comment:
UI composition, usually we use drivers for that

Reply:
Fair. However the Action implementations of this are used in is building its own viewmodel (for its own view). I've looked at using drivers, but the "cleanest" way I found for that would have required me to rewrite a lot of Orchard.Users' AdminController so the list of existing users was generated more like the list of contents (basically, displaying SummaryAdmin views for each user, rather than a flat "custom" table). Not that we shouldn't get on that someday...

Reply:
if/when we do, implementations of this interface can be called in one of those drivers, or we could get rid of them and replace them with drivers.